docs/source/task-launch.rst

```diff
@@ -84,71 +105,131 @@ We can submit tasks from other tasks.  This allows us to make decisions while
+       ...:     b_future = client.submit(fib, n - 2)
+       ...:     a, b = client.gather([a_future, b_future])
+       ...:     return a + b
 
-   In [3]: def fib(n):
-      ...:     if n < 2:
-      ...:         return n
-      ...:     else:
-      ...:         with worker_client() as c
-      ...:             a = c.submit(fib, n - 1)
-      ...:             b = c.submit(fib, n - 2)
-      ...:             a, b = c.gather([a, b])
-      ...:             return a + b
-      ...:
+    In [3]: client = Client()
+
+    In [4]: future = client.submit(fib, 10)
+
+    In [5]: future.result()
+    Out[5]: 55
+
+This can deadlock the scheduler if too many nodes request jobs at once. To
+correct that, we can use ``secede`` and ``rejoin``. These functions will remove
+and rejoin the current node from the cluster respectively.
+
+.. code-block:: python
+
+    In [1]: def fib(n):
+       ...:     if n < 2:
+       ...:         return n
+       ...:     client = get_client()
+       ...:     a_future = client.submit(fib, n - 1)
+       ...:     b_future = client.submit(fib, n - 2)
+       ...:     secede()
+       ...:     a, b = client.gather([a_future, b_future])
+       ...:     rejoin()
+       ...:     return a + b
+
+Using ``secede`` can possibly have the worker move to a different node if it's
+long running. Transferring state between nodes can be costly time-wise if
+there's a lot of internal state (e.g., with a large matrix).
+
+``worker_client``
+~~~~~~~~~~~~~~~~~~~~~~
+
+``worker_client`` is a convenience function to do this for you so that you
+don't have to pass around connection information.  However you must use this
+function ``worker_client`` as a context manager to ensure proper cleanup on the
+worker.
+
+.. code-block:: python
+
+    In [1]: from distributed import worker_client
 
-   In [4]: future = e.submit(fib, 100)
+    In [2]: import time
 
-   In [5]: future
-   Out[5]: <Future: status: finished, type: int, key: fib-7890e9f06d5f4e0a8fc7ec5c77590ace>
+    In [3]: def fib(n):
+       ...:     time.sleep(1)  # to simulate a long running function
+       ...:     if n < 2:
+       ...:         return n
+       ...:      with worker_client() as client:
+       ...:          a_future = client.compute(fib, n - 1)
+       ...:          b_future = client.compute(fib, n - 2)
+       ...:          a, b = client.gather([a_future, b_future])
+       ...:      return a + b
 
-   In [6]: future.result()
-   Out[6]: 354224848179261915075
+    In [4]: client = Client()
 
-This example is a bit extreme and spends most of its time establishing client
-connections from the worker rather than doing actual work, but does demonstrate
-that even pathological cases function robustly.
+    In [5]: future = client.submit(fib, 10)
 
+    In [6]: future.result()
+    Out[6]: 55
 
-Technical details
-~~~~~~~~~~~~~~~~~
+This allows you to spawn tasks that themselves act as potentially long-running
+clients, managing their own independent workloads.
```

Comment:
This is mostly the same in the cases above as well.  Worker_client doesn't do anything other than call get_client, secede, and rejoin. This is also exactly what dask.compute does.  You're right that it does send the "long-running" signal up, but this rarely affects real workloads.